airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -361,14 +367,32 @@ public WebBackendConnectionRead webBackendUpdateConnection(final WebBackendConne
     connectionRead = connectionsHandler.updateConnection(connectionUpdate);
 
     if (needReset) {
-      ManualOperationResult manualOperationResult = eventRunner.synchronousResetConnection(
-          webBackendConnectionUpdate.getConnectionId(),
-          // TODO (https://github.com/airbytehq/airbyte/issues/12741): change this to only get new/updated
-          // streams, instead of all
-          configRepository.getAllStreamsForConnection(webBackendConnectionUpdate.getConnectionId()));
-      verifyManualOperationResult(manualOperationResult);
-      manualOperationResult = eventRunner.startNewManualSync(webBackendConnectionUpdate.getConnectionId());
-      verifyManualOperationResult(manualOperationResult);
+      final UUID connectionId = webBackendConnectionUpdate.getConnectionId();
+      final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody().connectionId(connectionId);
+      final ConnectionStateType stateType = getStateType(connectionIdRequestBody);
+      final List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset;
+
+      if (stateType == ConnectionStateType.LEGACY || stateType == ConnectionStateType.NOT_SET || stateType == ConnectionStateType.GLOBAL) {
```

Comment:
I think that this implies that we can't do a partial reset for global. I think that this is not the expected behavior (i.e. : We need to calculate the diff for global as well)

Reply:
Are they producing a GLOBAL state? If they aren't compatible with the global state, they should produce a legacy state. I think that it is something that we might forget and become a future bug where the UI indicate that we will do a partial reset but in the end we will perform a full reset.

Reply:
@alovew I think that I now understand the issue which is that are producing a non legacy state where we should. Let's keep that in mind to not forget to remove.

Reply:
Yeah I believe the CDC sources will produce a GLOBAL state (@jdpgrailsdev to confirm?), so until CDC sources can support partial resets, we want to still reset everything.

@benmoriceau is the UI currently checking the state type and returning a message about partial resets for GLOBAL and STREAM? If so we may want to ask them to change that logic to only show the partial reset message for `STREAM` state type, until the CDC support for partial resets is in and we remove this `|| GLOBAL` here.

@alovew could you write a brief ticket and put it in our current sprint to remove this `|| GLOBAL` once CDC sources support partial resets? Just to make extra sure that we don't forget to do that

Reply:
@lmossman Correct.  The updated connector (Postgres) CDC version will produce both the new `GLOBAL` payload plus the legacy `data` payload, that has the legacy `CdcState` blob in it.

Reply:
made a ticket: https://app.zenhub.com/workspaces/oss-platform-625a01eb010e82001390f49e/issues/airbytehq/airbyte/14319